### PR TITLE
ui: Remove dependency on the scene package from the ui package

### PR DESCRIPTION
**Pull request by hajimehoshi**
This is necessary to make scene.Manager have a component in the
package ui.

### CODE REVIEW

File: internal/ui/titleview.go [hunk 1]
```diff
@@ -252,7 +266,7 @@ func (t *TitleView) drawTitle(screen *ebiten.Image) {
 }
 
 func (t *TitleView) Draw(screen *ebiten.Image) {
-	if !t.initialized {
+	if !t.initedUI {
```
Comment (wasedaigo) [inline internal/ui/titleview.go]: inited sounds not familiar with me

File: internal/ui/titleview.go [hunk 2]
```diff
@@ -48,51 +40,76 @@ type TitleView struct {
...
+func (t *TitleView) initUI() {
+	w, h := t.sceneWidth, t.sceneHeight
 
 	settingsIcon := assets.GetImage("system/common/icon_settings.png")
 	moreGamesIcon := assets.GetImage("system/common/icon_moregames.png")
 
 	by := 16
 	t.footerOffset = 0
-	if sceneManager.HasExtraBottomGrid() {
+	if t.sceneHeight > consts.SuperLargeScreenHeight {
```
Comment (wasedaigo) [inline internal/ui/titleview.go]: This part became less readable. If you want this change you should try to eliminate HasExtraBottomGrid() altogether..

**Comment by wasedaigo** [comment]
A bit more explanation is necessary, how cutting the dependency for sceneManager from TitleView helps?

File: internal/ui/titleview.go (hunk 1 above)
Comment (hajimehoshi) [reply internal/ui/titleview.go]: What is your suggestion? :)

File: internal/ui/titleview.go (hunk 2 above)
Comment (hajimehoshi) [reply internal/ui/titleview.go]: Good point. Done.

File: internal/ui/titleview.go (hunk 1 above)
Comment (wasedaigo) [reply internal/ui/titleview.go]: initialized

Comment (hajimehoshi) [reply internal/ui/titleview.go]: Done.

**Pull request merged**